src/ai/backend/manager/api/session.py

```diff
@@ -879,117 +884,109 @@ async def create_cluster(request: web.Request, params: Any) -> web.Response:
-                'creation_config': {
-                    'mount': mounts,
-                    'mount_map': mount_map,
-                    'environ': environ,
-                },
-            }
+    if _mounts := template['spec'].get('mounts'):  # noqa
+        mounts = list(_mounts.keys())
+        mount_map = {
+            key: value
+            for (key, value) in _mounts.items()
+            if len(value) > 0
+        }
+    if _environ := template['spec'].get('environ'):  # noqa
+        environ = _environ
+
+    log.debug('cluster template: {}', template)
+
+    kernel_configs: List[KernelEnqueueingConfig] = []
+    for node in template['spec']['nodes']:
+        # Resolve session template.
+        kernel_config = {
+            'image_ref': template['spec']['kernel']['image'],
+            'cluster_role': node['cluster_role'],
+            'creation_config': {
+                'mount': mounts,
+                'mount_map': mount_map,
+                'environ': environ,
+            },
+        }
 
-            if session_template['spec']['sess_type'] == 'interactive':
-                kernel_config['sess_type'] = SessionTypes.INTERACTIVE
-            elif session_template['spec']['sess_type'] == 'batch':
-                kernel_config['sess_type'] = SessionTypes.BATCH
-
-            # TODO: Remove `type: ignore` when mypy supports type inference for walrus operator
-            # Check https://github.com/python/mypy/issues/7316
-            # TODO: remove `NOQA` when flake8 supports Python 3.8 and walrus operator
-            # Check https://gitlab.com/pycqa/flake8/issues/599
-            if tag := session_template['metadata'].get('tag', None):
-                kernel_config['tag'] = tag
-            if runtime_opt := session_template['spec']['kernel']['run']:
-                if bootstrap := runtime_opt['bootstrap']:
-                    kernel_config['bootstrap_script'] = bootstrap
-                if startup := runtime_opt['startup_command']:
-                    kernel_config['startup_command'] = startup
-
-            if resources := template['spec'].get('resources'):
-                kernel_config['creation_config']['resources'] = resources
-
-            if git := session_template['spec']['kernel']['git']:  # noqa
-                if _dest := git.get('dest_dir'):  # noqa
-                    target = _dest
-                else:
-                    target = git['repository'].split('/')[-1]
+        if template['spec']['sess_type'] == 'interactive':
+            kernel_config['sess_type'] = SessionTypes.INTERACTIVE
+        elif template['spec']['sess_type'] == 'batch':
+            kernel_config['sess_type'] = SessionTypes.BATCH
+
+        # TODO: Remove `type: ignore` when mypy supports type inference for walrus operator
+        # Check https://github.com/python/mypy/issues/7316
+        # TODO: remove `NOQA` when flake8 supports Python 3.8 and walrus operator
+        # Check https://gitlab.com/pycqa/flake8/issues/599
```

Comment:
Though this is not related to, since this PR is changing this code block: both python/mypy#7316 and PyCQA/flake8#323 (migrated from Gitlab issue) is both resolved and released, I think it'll be good to remove all `#noqa` annotations.

Reply:
Good point! 👍🏼